cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -466,9 +478,7 @@ arrow::Status VeloxShuffleWriter::splitRowVector(const facebook::velox::RowVecto
 arrow::Status VeloxShuffleWriter::splitFixedWidthValueBuffer(const facebook::velox::RowVector& rv) {
   for (auto col = 0; col < fixedWidthColumnCount_; ++col) {
     auto colIdx = simpleColumnIndices_[col];
-    auto column = rv.childAt(colIdx);
-    assert(column->isFlatEncoding());
-
+    auto column = flatChildAt(rv, colIdx);
```

Comment:
Why `flatChildAt` becomes required in the patch?

Reply:
Because in `facebook::velox::BaseVector::flattenVector`, it loads lazy vectors but preserves lazy encoding.

Reply:
I see. Then is it possible to handle the case inside `VeloxColumnarBatch::ensureFlattened` ?